File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        total = 0
        left = 0
        min_len = float('inf')
        for right in range(len(nums)):
            total += nums[right]
            while total >= target:
                min_len = min(right - left + 1, min_len)
                total -= nums[left]
                left += 1
        return min_len if min_len != float('inf') else 0

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int total = 0;
        int left = 0;
        int minSize = Integer.MAX_VALUE;
        for(int right =0; right < nums.length;right++){
            total += nums[right];
            while (total >= target){
                minSize = Math.min(minSize, right-left+1);
                total -= nums[left];
                left++;
            }
            
        }
        return minSize == Integer.MAX_VALUE ? 0 : minSize;

    }
}